tests/test_set_config.py

```diff
@@ -306,3 +306,88 @@ def test_show_format():
         result = outstring.getvalue()
 
         assert result == outcome
+
+def test_values_needed_yaml():
+    '''Test that the values_needed flag logs variables provided, variables left as 
+    jinja2 templates, and values still needed'''
+
+    input_file = os.path.join(uwtools_file_base, pathlib.Path("fixtures/srw_example.yaml"))
+    args = ['-i', input_file, '--values_needed']
+
+    # Capture stdout for values_needed output
+    outstring = io.StringIO()
+    with redirect_stdout(outstring):
+        set_config.create_config_obj(args)
+    result = outstring.getvalue()
+    outcome=\
+    """Filled template variables:
+FV3GFS
```

Comment:
It seems like this one might show up as a section heading to tell the user which section to find the variable:

```
Variables left as jinja2 templates:
FV3GFS.nomads.protocol
```

This might help the user if we had other entries in this file like this:
```
RAP:
  nomads:
    url: '{{ some template }}'     
```

Reply:
We have a fix for this that still iterates through everything once (and keeps track of the variable path as it goes), but the linter is flagging it for too-many-locals (16/15) and too-many-arguments(6/5).  The alternatives seem to be iterating through the whole file again, or creating a list of tuples which would be bad for readability

Reply:
Hmmm. I was thinking it might be simpler than that. 

If you add an argument to the method like `parent=""`, then you could update that as you recurse through (using the syntax I suggested in the review)....

```
def iterate_values(self, set_var, jinja2_var, empty_var, parent=""):

        for key, val in self.data.items():
            if isinstance(val, dict):
                set_var.append(f'    {parent}{key}')
                self.iterate_values(val, set_var, jinja2_var, empty_var, parent=f"{parent}{key}.")

```